### PR TITLE
zlog: init at 1.2.12

### PR DESCRIPTION
**Pull request by matthiasbeyer**
My attempt to package zlog.

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Forgot metadata and so on, sorry.

Will add these things after lunch.

File: pkgs/development/libraries/zlog/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    mkdir -p $out
+    make PREFIX=$out
+  '';
+
+  installPhase = ''
+    make PREFIX=$out install
+  '';
+
+  meta = with stdenv.lib; {
+    description= "reliable, high-performance, thread safe, flexible, clear-model, pure C logging library.";
```
Comment (jagajaga) [inline pkgs/development/libraries/zlog/default.nix]: Description is out of rules..

Comment (matthiasbeyer) [inline pkgs/development/libraries/zlog/default.nix]: Unfortunately I cannot find the guide how the description should be formatted, sorry. Can you provide a pointer?

Comment (jagajaga) [inline pkgs/development/libraries/zlog/default.nix]: Oh, sure. I thought you know. https://nixos.org/nixpkgs/manual/#sec-standard-meta-attributes

Comment (matthiasbeyer) [inline pkgs/development/libraries/zlog/default.nix]: The problem is that these guides can be _everywhere_, literally. I searched the `CONTRIBUTING.md` and the wiki, but I forgot to look into the nixpkgs manual.

I will fix this, of course.

Comment (jagajaga) [inline pkgs/development/libraries/zlog/default.nix]: First point in https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#making-patches.

**Comment by matthiasbeyer** [comment]
I hope I got everything right now. Ping me if I can squash my commits!

File: pkgs/development/libraries/zlog/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, pkgs }:
```
Comment (bjornfor) [inline pkgs/development/libraries/zlog/default.nix]: It's generally preferred to not pull in the whole "pkgs" set, but instead list each needed dependency.

Comment (matthiasbeyer) [inline pkgs/development/libraries/zlog/default.nix]: Right. I don't even see a use of this variable, so I guess I can remove it completely. Hang on!

File: pkgs/development/libraries/zlog/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv, fetchurl }:
+
+stdenv.mkDerivation rec {
+  version = "1.2.12";
+  name = "zlog-${version}";
+
+  src = fetchurl {
+    url = "https://github.com/HardySimpson/zlog/archive/${version}.tar.gz";
```
Comment (bjornfor) [inline pkgs/development/libraries/zlog/default.nix]: Looks like a generated archive, better use fetchzip or fetchFromGitHub to not get hash mismatch in the future.

**Comment by matthiasbeyer** [comment]
Hoops. Fixed, thank you for pointing out!

**Comment by jagajaga** [comment]
I think it's ready for squashing.

**Comment by matthiasbeyer** [comment]
Squashed. We should wait until release-15.07 branch-off happened, I guess.

**Comment by bjornfor** [comment]
Pushed to master (20882964f518acb2aba364a25d03c96fd149821a). (It's OK to _add_ packages to the release branch, or to master, close to the release. But we should be very careful when _updating_ packages on/close to release.)

**Pull request closed without merging**